### ISSUE TITLE
Add an option for setup a local mirror just like "repo init --mirror"?

### ISSUE THREAD

**Issue by swordligit** [zephyrproject-rtos/west#658]
For android repo tool, there is an option 'repo init --mirror' to setup a local 
mirror, but with west,  have to git clone --bare all the projects one by one.

So, is it possible to add this option?
Br,
Yingchun

**Comment by mbolivar-nordic** [comment]
This seems like a valid way to set up a cache you could use with `west update --name-cache`. But since `west init` only clones the manifest repository, what would the commands that you would like to type be, and what would the desired outcomes be?

**Comment by swordligit** [comment]
In my local server, I want to clone all the repository with the bare mode,
then all my members can clone whole projects from my local server.
not from the upsteam.

Here is my simple steps now:
```
west init -m https://github.com/zephyrproject-rtos/zephyr

west list | awk '{print $4}' > projects.list
mkdir -p /path/of/mirror/; cd /path/of/mirror

for p in $(cat projects.list); do git clone --bare $p; done
```
After that, I can change the upstream url in west.yaml and download the projects
from my mirror.

I thought this is almost the android `repo init --mirror ` does.

**Comment by marc-hb** [comment]
```shell
west list -f '{url}' | while read -r loc; do echo git clone "$loc"; done
```

Related:
- #615

**Comment by swordligit** [comment]
seems there is a real requirement,can this be implemented? using an extra script is
not so convenient.

**Comment by marc-hb** [comment]
Requirement funded by whom?

**Comment by marc-hb** [comment]
There two ways to manage valid but unfunded requests in issue trackers.

1. The "project manager" way: close as wontfix because not funded.
2. The "developer" way: keep open because valid.

Most open-source projects work in the second way  because there is rarely any inter-company project management.

**Comment by mbolivar-nordic** [comment]
Agree, this is a valid enhancement and I would take a patch adding a new `west mirror` command that accomplished this. This doesn't belong in `west init` since that command's job is to set up a new workspace with just a manifest repository in it.

**Comment by thorsten-klein** [comment]
I think we’ll have a similar use case in the future.

However, I’m not sure if introducing a `west mirror` command makes sense, as this topic could get quite complex. 
For example, there can be multiple projects that share the same git repository name:
```
https://github.com/abc/module.git
https://github.com/xyz/module.git
```

When mirroring, users would need to define individual mirror URLs, since a repository like `module` can only exist once within their own GitHub organization. 
It’s up to each user how they want to handle that — for instance, by renaming repositories (`abc-library.git`, `xyz-library.git`) or placing them in separate organizations.

Instead of adding a dedicated `west mirror` command, I’d suggest enhancing `west list` to support outputting its data in a structured format, such as JSON. That way, users can easily post-process the information with their own scripts or tools as needed (or their own west `mirror` command extension).

**Comment by marc-hb** [comment]
@swordligit , there have been some caching enhancements added by @thorsten-klein recently. Can you upgrade west and check whether they would help in your use case?

**Comment by marc-hb** [comment]
@swordligit , please ask to re-open if you do not find the new caching features good enough for your use case(s).
